dags/glam_fenix_release.py

```diff
@@ -0,0 +1,119 @@
+from datetime import datetime, timedelta
+from functools import partial, reduce
+
+from airflow import DAG
+from airflow.operators.empty import EmptyOperator
+from airflow.sensors.external_task import ExternalTaskSensor
+from airflow.utils.task_group import TaskGroup
+
+from utils.constants import ALLOWED_STATES, FAILED_STATES
+from utils.glam_subdags.generate_query import (
+    generate_and_run_glean_task,
+)
+from utils.tags import Tag
+
+default_args = {
+    "owner": "efilho@mozilla.com",
+    "depends_on_past": False,
+    "start_date": datetime(2025, 1, 22),
+    "email": [
+        "telemetry-alerts@mozilla.com",
+        "akomarzewski@mozilla.com",
+        "efilho@mozilla.com",
+    ],
+    "email_on_failure": True,
+    "email_on_retry": True,
+    "retries": 2,
+    "retry_delay": timedelta(minutes=30),
+}
+
+PROJECT = "moz-fx-glam-prod"
+
+tags = [Tag.ImpactTier.tier_1]
+
+with DAG(
+    "glam_fenix_release",
+    default_args=default_args,
+    max_active_runs=1,
+    schedule_interval="0 10 * * 6",  # 10am on Saturday
+    doc_md=__doc__,
+    tags=tags,
+) as dag:
+    wait_for_glam_fenix = ExternalTaskSensor(
+        task_id="wait_for_daily_fenix_release",
+        external_dag_id="glam_fenix",
+        external_task_id="org_mozilla_fenix_glam_release_done",
+        execution_delta=timedelta(days=-5, hours=-16),
+        check_existence=True,
+        mode="reschedule",
+        allowed_states=ALLOWED_STATES,
+        failed_states=FAILED_STATES,
+        pool="DATA_ENG_EXTERNALTASKSENSOR",
+        email_on_retry=False,
+    )
+
+    for product in ["org_mozilla_fenix_glam_release"]:
+        func = partial(
+            generate_and_run_glean_task,
+            product=product,
+            destination_project_id=PROJECT,
+            env_vars={"STAGE": "incremental"},
+        )
+        view, init, query = (
+            partial(func, task_type=task_type)
+            for task_type in ["view", "init", "query"]
+        )
+
+        # stage 2 - downstream for export
```

Comment:
I think these task definitions could be put in a function and shared between the dags, which would make it easier to maintain and ensure they're identical. It's fine as-is though

Reply:
It's something I definitely intend to do